src/main/java/com/github/hokkaydo/eplbot/module/code/JavaRunner.java

```diff
@@ -0,0 +1,122 @@
+public class JavaRunner {
+    private static final String OUTPUT_PATH = System.getProperty("user.dir")+"\\src\\temp\\";
+    public static String run(String input) {
+        if (!input.equals(safeImports(input))){return "Unvalid imports";};
+        String class_name = regexClassName(input);
+        writeFile(input,Path.of(OUTPUT_PATH+"\\"+class_name+".java"));
+        String filePath = OUTPUT_PATH + "\\" + class_name + ".java";
+        try {
+            ByteArrayOutputStream error_stream = new ByteArrayOutputStream();
+            if (ToolProvider.getSystemJavaCompiler().run(null, null, new PrintStream(error_stream), filePath) == 0) {
+                try {
+                    Process process = new ProcessBuilder(System.getProperty("java.home") + File.separator + "bin" + File.separator + "java", "-cp", System.getProperty("java.class.path") + File.pathSeparator + OUTPUT_PATH,regexClassName(input) ).redirectErrorStream(true).start();
+                    Thread timeoutThread = new Thread(() -> {
+                        try {
+                            Thread.sleep(1000*Integer.parseInt(Strings.getString("COMMAND_CODE_TIMELIMIT")));
+                            process.destroy();
+                        } catch (InterruptedException e) {}
+                    });
+                    timeoutThread.start();
+                    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                    byte[] buffer = new byte[1024];
+                    int bytesRead;
+                    while ((bytesRead = process.getInputStream().read(buffer)) != -1) {
+                        outputStream.write(buffer, 0, bytesRead);
+                    }
+                    if (process.waitFor() == 0) {
+                        timeoutThread.interrupt();
+                        for (File file : new File(OUTPUT_PATH).listFiles()) {
+                            if (file.isFile() && file.getName().startsWith(class_name)) {
+                                file.delete();
+                            }                            
+                        }
+                    return outputStream.toString();
+                    } else {
+                        timeoutThread.interrupt();
+                        for (File file : new File(OUTPUT_PATH).listFiles()) {
+                            if (file.isFile() && file.getName().startsWith(class_name)) {
+                                file.delete();
+                            }                            
+                        }
+                        String output = outputStream.toString().trim();
+                        if (output.isEmpty()) {
+                            return "Run failed: Timelimit exceeded "+Strings.getString("COMMAND_CODE_TIMELIMIT")+" s";
+                        } else {
+                            return "Run failed:\n" + output;
+                        }
+                    }
+                } catch (IOException | InterruptedException e) {
```

Comment:
```suggestion
                } catch (IOException e) {
```
You already caught the interruption